skrifa/src/scale/colr/instance.rs

```diff
@@ -0,0 +1,653 @@
+//! COLR table instance.
```

Comment:
What about putting this in `skrifa/src/colr/instance.rs`? Or `skrifa/src/color/instance.rs` - My thinking is: We're not really scaling here, mostly applying variations. And then `ColorPainter` can go in the same directory when it is ready.

Reply:
This seems reasonable but the details are a bit tricky. I’m thinking forward to adding bounding box computation which will require the scaling code. As of now, the “metadata level” types don’t have any dependencies on the scale module and I find that to be a clean separation. I suppose we can maintain that by just taking a closure for computing per glyph bounding boxes. Let me put together a quick proposal for this API.

Reply:
#702 is a proposal for this that tries to fit it into the current design for metadata access. I see your point about the matrix library, but it seems like `ColorPainter` needs the struct regardless as a parameter for transform callbacks. I believe your code already has the constructors for translate/rotate/skew, so we'd really only need to add `mat * mat` and `mat * point` operators to make this work.